lxd/device/usb.go

```diff
@@ -0,0 +1,230 @@
+package device
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/lxc/lxd/lxd/device/config"
+	"github.com/lxc/lxd/lxd/instance"
+	"github.com/lxc/lxd/shared"
+)
+
+type usb struct {
+	deviceCommon
+}
+
+// validateConfig checks the supplied config for correctness.
+func (d *usb) validateConfig() error {
+	if d.instance.Type() != instance.TypeContainer {
+		return ErrUnsupportedDevType
+	}
+
+	rules := map[string]func(string) error{
+		"vendorid":  shared.IsAny,
```

Comment:
We should add a validator for that and shared it with GPU too.
It's effectively 4 digit hex lowercase (unless we do case insensitive matches in which case we can allow both cases).

Reply:
Added a validator to usb and gpu.